beater/connections_test.go

```diff
@@ -88,23 +126,38 @@ func TestFilterConnectionsAssociatedWithListeningSockets(t *testing.T) {
 
 	go filterAndPublish(true, true, true, 5*time.Second, input, connections, servers)
 
-	input <- listeningConnection(80)
+	localIP := randIP()
+
+	input <- listeningConnectionOn(80, localIP)
 	_, ok := <-servers
 	assert.Equal(t, ok, true, "a server should be reported")
-	input <- incomingConnection(80)
+	input <- incomingConnectionTo(80, localIP)
 
 	time.Sleep(100 * time.Millisecond)
 
 	select {
 	case <-connections:
-		t.Fail()
+		t.Fatal("The incoming connection on the known local IP should not be reported")
 	case <-servers:
-		t.Fail()
+		t.Fatal("No server notification expected")
 	default:
 		// Nothing to read: OK!
 	}
 }
 
+func TestDeduplicateLocalIps(t *testing.T) {
+	input := make(chan *sockets.SocketInfo, 0)
+	connections, servers := make(chan Connection, 0), make(chan ServerConnection, 0)
+
+	go filterAndPublish(true, true, true, 5*time.Second, input, connections, servers)
+
+	input <- listeningConnection(80)
+	_, ok := <-servers
+	assert.Equal(t, ok, true, "a server should be reported")
+	input <- listeningConnection(80)
```

Comment:
Shouldnt there be some assert/test after the second listeningConnection(80)? I am not sure what you want to test otherwise... but perhaps I am missing something..